docs/src/test-api/class-testconfig.md

```diff
@@ -595,7 +595,7 @@ export default defineConfig({
 - type: ?<[Object]|[Array]<[Object]>>
   - `command` <[string]> Shell command to start. For example `npm run start`..
   - `port` ?<[int]> The port that your http server is expected to appear on. It does wait until it accepts connections. Exactly one of `port` or `url` is required.
-  - `url` ?<[string]> The url on your http server that is expected to return a 2xx, 3xx, 400, 401, 402, or 403 status code when the server is ready to accept connections. Exactly one of `port` or `url` is required.
+  - `url` ?<[string]> The url on your http server that is expected to return a 2xx, 3xx, 400, 401, 402, or 403 status code when the server is ready to accept connections. Redirects (3xx status codes) are being followed and the new location is checked. Exactly one of `port` or `url` is required.
```

Comment:
I like the simple formula that we currently document:

```
The url on your http server that is expected to return a
2xx, 3xx, 400, 401, 402, or 403 status code when the
server is ready to accept connections.
```

Does that not work? We can prevent probe from following the redirects if current behavior does not much the doc above.

Reply:
Yeah, I mentioned that in the PR description: At the moment, the docs don’t match reality: redirects are _attempted_ to be followed, but it’s buggy. I am fine with both not following redirects, or fixing redirects. But not following redirects is more work, because the underlying utility (httpRequest) does not support that yet. And I wanted to keep things as simple as possible. But if that’s the preferred solution, I’m also Ok with that. I’d probably need some guidance on how to add that „don’t follow redirects“ option to `httpRequest`.

Reply:
https://github.com/microsoft/playwright/blob/main/packages/playwright-core/src/utils/network.ts#L78 we are using http.request for that, which should not follow redirects. Could you add a failing tests that surfaces the bug?

Reply:
`http.request` may not follow redirects, but there is dedicated code in `httpRequest` to do so: https://github.com/microsoft/playwright/blob/main/packages/playwright-core/src/utils/network.ts#L73

But this code is broken, which is the whole point of this PR. At the moment it fixes the broken redirect implementation. But I’m also happy to remove the redirect behavior, if that’s preferred.

Regarding a test case that fails, here is the simplest one I can think of:
```ts

test('should follow redirects', async ({ runInlineTest, server }) => {
  server.setRedirect('/redirect', '/redirected-to');
  const result = await runInlineTest({
    'test.spec.ts': `
      import { test, expect } from '@playwright/test';
      test('connect to the server', async () => {
        expect(true).toBe(true);
      });
    `,
    'playwright.config.ts': `
      module.exports = {
        webServer: {
          command: 'exit 1',
          url: 'http://localhost:${server.PORT}/redirect',
          reuseExistingServer: true,
        }
      };
    `,
  });
  expect(result.exitCode).toBe(0);
});
```